stable/redis/templates/pvc.yaml

```diff
@@ -9,15 +9,24 @@ metadata:
     release: "{{ .Release.Name }}"
     heritage: "{{ .Release.Service }}"
   annotations:
+  {{- if lt .Capabilities.KubeVersion.Minor "7" }}
   {{- if .Values.persistence.storageClass }}
     volume.beta.kubernetes.io/storage-class: {{ .Values.persistence.storageClass | quote }}
   {{- else }}
     volume.alpha.kubernetes.io/storage-class: default
   {{- end }}
+  {{- end }}
 spec:
   accessModes:
     - {{ .Values.persistence.accessMode | quote }}
   resources:
     requests:
       storage: {{ .Values.persistence.size | quote }}
+  {{- if ge .Capabilities.KubeVersion.Minor "7" }}
+  {{- if .Values.persistence.storageClass }}
```

Comment:
Going forward, this is the approach we should follow: https://github.com/kubernetes/charts/pull/1332/files#diff-b1da1e98ff7b4dd399a177788866ddbaR20

Specifically:
- we set storageClassName only if storageClass value is defined
- a value of "-" allows the storageClassName to be set to an empty string (empty string disables dynamic provisioning in Kubernetes)
- when storageClass is not set, we also don't set storageClassName (this has the effect of Kubernetes picking the default storage class automatically).

A value of `storageClassName: default` is not sufficient for the last case because the default storage class is named differently on different Kubernetes platforms (e.g. standard in some cases).

Reply:
If you create a PVC with an empty string for the `storageClassName` - the pvc is still created, it just stays pending forever.  I'd rather that we never create the pvc in that case - by wrapping the pvc record in an if statement.